sdk/dotnet/Pulumi/Testing/TestResult.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2016-2020, Pulumi Corporation
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace Pulumi.Testing
+{
+    /// <summary>
+    /// Represents an outcome of a test run.
+    /// </summary>
+    public class TestResult
+    {
+        /// <summary>
+        /// All Pulumi resources that got registered during the run.
+        /// </summary>
+        public ImmutableArray<Resource> Resources { get; }
+        
+        /// <summary>
+        /// Outputs from the stack.
+        /// </summary>
+        public ImmutableDictionary<string, object> StackOutputs { get; }
+
+        /// <summary>
+        /// Extract a value from an <see cref="Output{T}"/>. Can only be used
+        /// for outputs fount in <see cref="Resources"/>.
+        /// </summary>
+        /// <param name="output">An output containing a value.</param>
+        /// <returns>An awaitable task that returns the value.</returns>
+        public Task<T> GetAsync<T>(Output<T> output) => output.GetValueAsync();
```

Comment:
Why do we need/want to expose this here?

Reply:
Basically, to assert output values. E.g.
```cs
var consumptionPlan = result.Resources.OfType<Plan>().Single();
var kind = await result.GetAsync(consumptionPlan.Kind);
kind.Should().Be("FunctionApp");
```
Do you have better ideas?

Reply:
Removed this to move the PR to completion. Need to refine the pattern while writing docs.